kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaRestoreService.java

```diff
@@ -0,0 +1,160 @@
+package com.slack.kaldb.clusterManager;
+
+import static com.slack.kaldb.clusterManager.ReplicaCreationService.replicaMetadataFromSnapshotId;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.slack.kaldb.metadata.replica.ReplicaMetadata;
+import com.slack.kaldb.metadata.replica.ReplicaMetadataStore;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadata;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+import javax.naming.SizeLimitExceededException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Scheduled service responsible for queueing and creating replica snapshots */
+public class ReplicaRestoreService extends AbstractScheduledService {
+  private ScheduledFuture<?> pendingTask;
+  private final KaldbConfigs.ManagerConfig managerConfig;
+  private final KaldbConfigs.ManagerConfig.ReplicaRestoreServiceConfig replicaRestoreServiceConfig;
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private final BlockingQueue<SnapshotMetadata> queue;
+  private final ReplicaMetadataStore replicaMetadataStore;
+  private final MeterRegistry meterRegistry;
+  private final Counter successfullyCreatedReplicas;
+  private final Counter failedReplicas;
+  private final Counter skippedReplicas;
+  private final Timer restoreTimer;
+  protected static final Logger LOG = LoggerFactory.getLogger(ReplicaCreationService.class);
+  public static String REPLICAS_CREATED = "replicas_created";
+  public static String REPLICAS_FAILED = "replicas_failed";
+  public static String REPLICAS_SKIPPED = "replicas_skipped";
+  public static String REPLICAS_RESTORE_TIMER = "replicas_restore_timer";
+
+  public ReplicaRestoreService(
+      ReplicaMetadataStore replicaMetadataStore,
+      MeterRegistry meterRegistry,
+      KaldbConfigs.ManagerConfig managerConfig) {
+    this.managerConfig = managerConfig;
+    this.replicaRestoreServiceConfig = managerConfig.getReplicaRestoreServiceConfig();
```

Comment:
I'm slightly confused as to why we set both the manager config, and then a subset of the manager config to their own variables. We probably only need one or the other here.